set.mm

```diff
@@ -615339,6 +615436,30 @@ number axioms (add ~ ax-10 , ~ ax-11 , ~ ax-13 , ~ ax-nul , and remove
     VEWMTWGTWDEEAVSWEWFWONNJWETWFTVRGAWDERQBWNVDKVREGWDEWNNNWDTVFSVGSAEBHCVHUJV
     IVJVKIVLUPUQUNOURVM $.
 
+  ${
+    oexpreposd.n $e |- ( ph -> N e. RR ) $.
+    oexpreposd.m $e |- ( ph -> M e. NN ) $.
+    oexpreposd.1 $e |- ( ph -> -. ( M / 2 ) e. NN ) $.
+    $( Lemma for ~ dffltz .  (Contributed by Steven Nguyen, 4-Mar-2023.) $)
+    oexpreposd $p |- ( ph -> ( 0 < N <-> 0 < ( N ^ M ) ) ) $=
```

Comment:
I haven't looked at how this is used, but it seems more natural to state
```
oexpreposd.m $e |- ( ph -> M e. NN0 ) $.
oexpreposd $p |- ( ph -> ( 0 < N <-> 0 < ( N ^ ( ( 2 x. M ) + 1 ) ) ) ) $=
```

Reply:
It's used in dffltz, here's the (limitedly useful) generate webpage -> print but save as pdf visual https://github.com/icecream17/Stuff/blob/master/dffltz.pdf

Basically the way I proved this is to set `x = if ( ( n / 2 ) e. NN , ( abs ` a ) , if ( 0 < a , if ( 0 < b , a , if ( 0 < c , -u b , a ) ) , if ( 0 < b , if ( 0 < c , -u a , b ) , -u a ) ) ), y = ..., z = ...`. and then splitting out the if statements using theorems like `ph /\ ps -> ch` and `ph /\ -. ps -> ch` and collapsing the if statements per branch using iftrue and iffalse.

It was convenient to have the ( n / 2 ) e. NN for a ^ ( 2 * ( n / 2 ) ) purposes... but I did notice how many theorems were of a different format and how oexpreposd is way longer than it should be. And while checking I just noticed nneo (!) which at least replaces that last hypothesis with `( ( M + 1 ) / 2 ) e. NN`. In the case of the current x = ... y = ... z = ... in dffltz though, this would just add an extra conversion step (the first condition is `( n / 2 ) e. NN` which is very similar to df-even).

As a sidenote, both of these are apparently not the current even/odd convention...
https://github.com/metamath/set.mm/issues/2433